classes/institution/Collector.inc.php

```diff
@@ -0,0 +1,174 @@
+<?php
+/**
+ * @file classes/institution/Collector.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class institution
+ *
+ * @brief A helper class to configure a Query Builder to get a collection of institutions
+ */
+
+namespace PKP\institution;
+
+use Illuminate\Database\Query\Builder;
+use Illuminate\Support\Facades\DB;
+use PKP\core\interfaces\CollectorInterface;
+
+class Collector implements CollectorInterface
+{
+    public DAO $dao;
+    public ?array $contextIds = null;
+    public ?array $ips = null;
+    public string $searchPhrase = '';
+    public ?int $count = null;
+    public ?int $offset = null;
+    public bool $includeSoftDeletes = false;
+
+    public function __construct(DAO $dao)
+    {
+        $this->dao = $dao;
+    }
+
+    /**
+     * Filter institutions by one or more contexts
+     */
+    public function filterByContextIds(?array $contextIds): self
+    {
+        $this->contextIds = $contextIds;
+        return $this;
+    }
+
+    /**
+     * Filter institutions by one or more IPs
+     */
+    public function filterByIps(?array $ips): self
+    {
+        $this->ips = $ips;
+        return $this;
+    }
+
+    /**
+     * Filter institutions by those matching a search query
+     */
+    public function searchPhrase(string $phrase): self
+    {
+        $this->searchPhrase = $phrase;
+        return $this;
+    }
+
+    /**
+     * Limit the number of objects retrieved
+     */
+    public function limit(?int $count): self
+    {
+        $this->count = $count;
+        return $this;
+    }
+
+    /**
+     * Offset the number of objects retrieved, for example to
+     * retrieve the second page of contents
+     */
+    public function offset(?int $offset): self
+    {
+        $this->offset = $offset;
+        return $this;
+    }
+
+    /**
+     * Consider soft deleted institutions
+     */
+    public function includeSoftDeletes(bool $include = true): self
+    {
+        $this->includeSoftDeletes = $include;
+        return $this;
+    }
+
+    /**
+     * @copydoc CollectorInterface::getQueryBuilder()
+     */
+    public function getQueryBuilder(): Builder
+    {
+        $qb = DB::table($this->dao->table . ' as i');
```

Comment:
Just a preference, but I prefer to add `->select('i.*')` here. Then anything that needs to augment this select (e.g. for a plugin or something) can use `addSelect` to add columns. If other code needs to override it (e.g. the calls that just get the IDs) they can replace the column list by just calling `->select(...)` again.